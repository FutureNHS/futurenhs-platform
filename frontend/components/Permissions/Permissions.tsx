import React, { HTMLProps, useState } from "react";

import { FormElementProps } from "nhsuk-react-components/lib/util/types/FormTypes";
import styled from "styled-components";

import { ChevronBottomIcon, ChevronTopIcon } from "../Icon";
import PermissionsRadio from "../PermissionsRadio/PermissionsRadio";

interface PermissionsProps
  extends HTMLProps<HTMLInputElement>,
    FormElementProps {
  inputRef?: (inputRef: HTMLInputElement | null) => any;
}

const TitleContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin-bottom: 24px;

  h3 {
    margin: 0;
  }

  button {
    margin: 0 0 0 20px;
    background: none;
    color: inherit;
    border: none;
    outline: inherit;
    border-top: 4px solid transparent;
    border-bottom: 4px solid transparent;
    cursor: pointer;

    :focus {
      background-color: ${({ theme }) => theme.colorNhsukYellow};
      border-bottom: 4px solid ${({ theme }) => theme.colorNhsukBlack};
    }
  }
`;

const Description = styled.p`
  margin-bottom: 40px;
`;

const Permissions: React.FC<PermissionsProps> = ({ inputRef }) => {
  const [expanded, setExpandedState] = useState(false);
  return (
    <>
      <TitleContainer>
        <h3>Permissions</h3>
        <button
          type="button"
          data-cy="permissions-button"
          onClick={() => setExpandedState(!expanded)}
        >
          {expanded ? (
            <ChevronTopIcon title={`Hide permissions`} />
          ) : (
            <ChevronBottomIcon title={`Show permissions`} />
          )}
        </button>
      </TitleContainer>
      <Description>
        You can set permissions for this folder. All users authorised to access
        a folder can view all its contents
      </Description>
      <PermissionsRadio hidden={!expanded} inputRef={inputRef} />
    </>
  );
};

export default Permissions;
