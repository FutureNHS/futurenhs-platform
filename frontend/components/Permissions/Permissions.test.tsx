import React from "react";

import { render } from "@testing-library/react";

import Permissions from "./Permissions";

describe(Permissions, () => {
  it("renders matching snapshot", () => {
    const container = render(<Permissions />);

    expect(container.asFragment()).toMatchSnapshot();
  });
});
