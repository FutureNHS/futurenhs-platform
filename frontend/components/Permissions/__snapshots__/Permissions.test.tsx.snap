// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permissions renders matching snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 24px;
}

.c0 h3 {
  margin: 0;
}

.c0 button {
  margin: 0 0 0 20px;
  background: none;
  color: inherit;
  border: none;
  outline: inherit;
  border-top: 4px solid transparent;
  border-bottom: 4px solid transparent;
  cursor: pointer;
}

.c0 button:focus {
  border-bottom: 4px solid;
}

<div
    class="c0"
  >
    <h3>
      Permissions
    </h3>
    <button
      data-cy="permissions-button"
      type="button"
    >
      <div
        class="icon-wrapper "
      >
        <svg
          aria-labelledby="chevronBottomIconId chevronBottomIconDesc"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="chevronBottomIconId"
          >
            Show permissions
          </title>
          <path
            clip-rule="evenodd"
            d="M11.5 16.5C11.1 16.5 10.8333 16.3714 10.5667 16.1143L3.9 9.68571C3.36667 9.17143 3.36667 8.4 3.9 7.88571C4.43333 7.37143 5.23333 7.37143 5.76667 7.88571L11.5 13.4143L17.2333 7.88571C17.7667 7.37143 18.5667 7.37143 19.1 7.88571C19.6333 8.4 19.6333 9.17143 19.1 9.68571L12.4333 16.1143C12.1667 16.3714 11.9 16.5 11.5 16.5V16.5Z"
            fill="#425462"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </button>
  </div>
  .c0 {
  margin-bottom: 40px;
}

<p
    class="c0"
  >
    You can set permissions for this folder. All users authorised to access a folder can view all its contents
  </p>
  .c0 .nhsuk-radios__label,
.c0 .nhsuk-radios__hint {
  padding-left: 28px;
  padding-top: 0px;
}

<div
    hidden=""
  >
    <h4>
      Viewing permissions
    </h4>
    <div
      class="nhsuk-form-group"
    >
      <div
        class="nhsuk-form-group"
      >
        <div
          class="nhsuk-radios c0"
          id="roleRequired"
          role="radiogroup"
        >
          <div
            class="nhsuk-radios__item"
          >
            <input
              aria-describedby="all-members--hint"
              aria-labelledby="all-members--label"
              checked=""
              class="nhsuk-radios__input"
              id="all-members"
              name="roleRequired"
              type="radio"
              value="PLATFORM_MEMBER"
            />
            <label
              class="nhsuk-label nhsuk-radios__label"
              for="all-members"
              id="all-members--label"
            >
              Make visible to all FutureNHS members
            </label>
            <span
              class="nhsuk-hint nhsuk-radios__hint"
              id="all-members--hint"
            >
              This will make the folder and its contents visible to all members of FutureNHS
            </span>
          </div>
          <div
            class="nhsuk-radios__item"
          >
            <input
              aria-describedby="workspace-members--hint"
              aria-labelledby="workspace-members--label"
              class="nhsuk-radios__input"
              id="workspace-members"
              name="roleRequired"
              type="radio"
              value="WORKSPACE_MEMBER"
            />
            <label
              class="nhsuk-label nhsuk-radios__label"
              for="workspace-members"
              id="workspace-members--label"
            >
              Make visible to all members of this workspace
            </label>
            <span
              class="nhsuk-hint nhsuk-radios__hint"
              id="workspace-members--hint"
            >
              This will make the folder and its contents visible to members of your workspace.
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
